File: adapter-binding/src/main/java/com/azazellj/adapter/binding/BindingViewHolder.java
package com.azazellj.adapter.binding;

import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.view.View;

/**
 * Created by azazellj on 10/2/17.
 */

public class BindingViewHolder
        extends com.azazellj.rv.adapter.common.AdapterViewHolder {
    /**
     * Default constructor.
     *
     * @param itemView view
     */
    public BindingViewHolder(View itemView) {
        super(itemView);
    }

    /**
     * Get holder binding.
     *
     * @param <T> holder binding class
     * @return binding
     */
    public <T extends ViewDataBinding> T getView() {
        return DataBindingUtil.getBinding(itemView);
    }
}

File: adapter-binding-kt/src/main/java/com/azazellj/recyclerview/adapter/binding/kt/BindingViewHolder.kt
package com.azazellj.recyclerview.adapter.binding.kt

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding

/**
 * Created by azazellj on 3/7/18.
 */
open class BindingViewHolder
/**
 * Default constructor.
 *
 * @param itemView view
 */(itemView: View) : com.azazellj.recyclerview.adapter.kt.common.AdapterViewHolder(itemView) {

    constructor(parent: ViewGroup, layoutId: Int) : this(getView(parent, layoutId))

    /**
     * Get holder binding.
     *
     * @param <T> holder binding class
     * @return binding
     */
    fun <T : ViewDataBinding> getView(): T? {
        return DataBindingUtil.getBinding(itemView)
    }

    companion object {
        fun getView(parent: ViewGroup, layoutId: Int): View {
            return getView(parent, layoutId, false)
        }

        fun getView(parent: ViewGroup, layoutId: Int, attachToParent: Boolean): View {
            return getView(LayoutInflater.from(parent.context), layoutId, parent, attachToParent)
        }

        fun getView(inflater: LayoutInflater, layoutId: Int, parent: ViewGroup, attachToParent: Boolean): View {
            return DataBindingUtil.inflate<ViewDataBinding>(inflater, layoutId, parent, attachToParent).root
        }
    }
}